src/tennissimulator/menu/Menu.java

```
@@ -119,32 +135,21 @@ private void selectPlayersForMatchSimulation() {
             return;
         }
 
-        Player firstPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(firstPlayer)) {
+        Optional<Player> firstPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        if (firstPlayer.isEmpty()) {
```

Here you are doing validation. Few lines below is done the same validation just with other player. Looks like duplicate code. What is more, I think you should introduce and throw an exception if player is not found. That way you wouldn't need to return from method, and later on in simulateMatch check if match is null would not be needed as well because if application managed to get there without exception it would mean that match is definitely not null. You would simply need to catch it in appropriate place and then application would continue working as expected.

---

Introduced Runtime Exception, which is caught in `simulateMatch()` method.